feature/gribi/otg_tests/vrf_policy_driven_te/README.md

```diff
@@ -0,0 +1,837 @@
+
+    ```
+
+2.  Apply vrf selection policy `vrf_selection_policy_w` to DUT port-1.
+
+3.  Send the following 6in4 and 4in4 flows to DUT port-1:
+
+    ```
+    * inner_src: `ipv4_inner_src`
+    * inner_dst: `ipv4_inner_encap_match`
+    * dscp: `dscp_encap_no_match`
+    * outter_src: `ipv4_outter_src_111`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_no_match`
+    * proto: `4`
+
+    * inner_src: `ipv6_inner_src`
+    * inner_dst: `ipv6_inner_encap_match`
+    * dscp: `dscp_encap_no_match`
+    * outter_src: `ipv4_outter_src_111`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_no_match`
+    * proto: `41`
+    ```
+
+4.  Verify that the packets have their outer v4 header stripped and are forwarded out of DUT port-8 per the BGP-ISIS routes in the DEFAULT VRF.
+
+5.  Verify that the TTL value is copied from the outer header to the inner header.
+
+6.  Change the subnet mask from /24 and repeat the test for the masks  /32, /22, and /28 and verify again that the packets are decapped and forwarded correctly.
+
+7.  Repeat the test with packets with a destination address that does not match the decap entry, and verify that such packets are not decapped.
+
+#### Test-2, match on source, protocol and DSCP, VRF_DECAP hit -> VRF_ENCAP_A miss -> DEFAULT
+
+1.  Using gRIBI to install the following entries in the `DECAP_TE_VRF`:
+
+    ```
+    IPv4Entry {192.51.100.1/24 (DECAP_TE_VRF)} -> NHG#50 (DEFAULT VRF) -> {
+        {NH#50, DEFAULT VRF, weight:1}
+    }
+    NH#50 -> {
+        decapsulate_header: OPENCONFIGAFTTYPESDECAPSULATIONHEADERTYPE_IPV4
+    }
+    ```
+    Install an 0/0 static route in TE_VRF_111
+
+    ```
+2.  Apply vrf selection policy `vrf_selection_policy_w` to DUT port-1.
+
+3.  Send the following 6in4 and 4in4 flows to DUT port-1:
+
+    ```
+    * inner_src: `ipv4_inner_src`
+    * inner_dst: `ipv4_inner_encap_no_match`
+    * dscp: `dscp_encap_a_1`
+    * outter_src: `ipv4_outter_src_111`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_a_1`
+    * proto: `4`
+
+    * inner_src: `ipv6_inner_src`
+    * inner_dst: `ipv6_inner_encap_no_match`
+    * dscp: `dscp_encap_a_1`
+    * outter_src: `ipv4_outter_src_111`
+    * outter_dst: `ipv4_outter_decap_match`
+    * dscp: `dscp_encap_a_1`
+    * proto: `41`
+    ```
+
+4.  Verify that the packets have their outer v4 header stripped and are forwarded out of DUT port-8 per the BGP-ISIS routes in the DEFAULT VRF.
+
+5.  Verify that the TTL value is copied from the outer header to the inner header.
+
+6.  Change the subnet mask from /24 and repeat the test for the masks  /32, /22, and /28 and verify again that the packets are decapped and forwarded correctly.
+
+#### Test-3, Mixed Prefix Decap gRIBI Entries
+
+Support for decap actions with mixed prefixes installed through gRIBI
+
+1.  Add the following gRIBI entries:
+
+    ```
+    IPv4Entry {192.51.129.0/22 (DECAP_TE_VRF)} -> NHG#52 (DEFAULT VRF) -> {
```

Comment:
the /22 prefix here should be 192.51.128.0/22 ?

Reply:
Fixed.